[Kenneth Wong Cun Wi] iP

Kurokishi592 opened the pull request.

Kurokishi592 committed:
add level 0: personalise chatbot name and implement skeletal version

Kurokishi592 committed:
add level 0: personalise chatbot name and implement skeletal version

Kurokishi592 committed:
add level 0: personalise chatbot name and implement skeletal version

Kurokishi592 committed:
Add level-1: echo feature and exit upon 'bye' input

Kurokishi592 committed:
Add level 2: the ability to store whatever text entered by the user and display them back to the user when requested.

Kurokishi592 committed:
Add level 3: Add the ability to mark tasks as done and change the status back to not done.

Kurokishi592 committed:
Add level 3: Add the ability to mark tasks as done and change the status back to not done.

Kurokishi592 committed:
Tweak the code to comply with good Java coding standard given

irw9n commented on src/main/java/Kurokishi.java:
```diff
@@ -1,18 +1,19 @@
         String name = "Kurokishi";
         String dashLine ="------------------------------------------------------------";
-        String promptCommand = "Please type something:) ";
+        String promptCommand = ("Simply type your task to add to a list\n" +
+                        "Type 'list' to see your tasks\n" +
+                        "Type 'mark <task number>' to mark a task as done\n" +
+                        "Type 'unmark <task number>' to unmark a task as not done\n" +
+                        "Type 'bye' to exit the program\n"
```
Very thoughtful to include a list of instructions for the user upon launch!

irw9n commented on src/main/java/Kurokishi.java:
```diff
@@ -35,33 +36,58 @@ public static void main(String[] args) {
+        System.out.println("Hello! I'm " + name +"\n" +
+                "I can help you keep track of your tasks!\n");
         System.out.println(promptCommand);
+        System.out.println(dashLine);
```
Good use of string variables to avoid repetition

asytrix99 replied:
Good use of user guide with `promptCommand`

irw9n commented on src/main/java/Kurokishi.java:
```diff
@@ -35,33 +36,58 @@ public static void main(String[] args) {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < taskIndex; i++) {
+                        System.out.println((i + 1) + ". " + taskList[i]);
+                    }
```
Good use of Egyptian style brackets!

irw9n commented on src/main/java/Kurokishi.java:
```diff
@@ -35,33 +36,58 @@ public static void main(String[] args) {
+                    }
+                }
+                else if (input.startsWith("mark ") || input.startsWith("unmark ")) {
+                    String[] parts = input.split(" ");
```
Good declaration of String[] array

irw9n reviewed (commented):
Overall, your code followed the Java coding standards well. LGTM.

Kurokishi592 committed:
Clean up the code for better readibility

Kurokishi592 committed:
Add Level-4 with Inheritance: Todo, Deadlines, Events tracking

asytrix99 commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,151 @@
+                        break;
+                    }
+                    int taskNumber = Integer.parseInt(parts[1]) - 1; // index zero stuff
+                    if (taskNumber >= 0 && taskNumber < taskIndex) {
+                        boolean isMark = command.equals("mark");
+                        taskList[taskNumber].setDone(isMark);
+                        if (isMark) {
+                            System.out.println("    Nice! I've marked this task as done:");
+                        } else {
+                            System.out.println("    OK, I've marked this task as not done yet:");
+                        }
+                        System.out.println("    " + taskList[taskNumber]);
+                    } else {
+                        System.out.println("    Invalid task number.");
+                    }
```
Maybe you could perform astraction here, can try to avoid arrowhead style code

asytrix99 commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,151 @@
+                        boolean isMark = command.equals("mark");
+                        taskList[taskNumber].setDone(isMark);
+                        if (isMark) {
+                            System.out.println("    Nice! I've marked this task as done:");
```
Can refactor this into a constant

asytrix99 commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,151 @@
+                        if (isMark) {
+                            System.out.println("    Nice! I've marked this task as done:");
+                        } else {
+                            System.out.println("    OK, I've marked this task as not done yet:");
```
Maybe you could refactor each print statement into constants? eg. 
`"    Nice! I've marked this task as done:"` can be refactored to `MARKED_DONE`

asytrix99 commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,151 @@
+                    break;
+
+                case "deadline":
+                    if (parts.length < 2 || !parts[1].contains(" /by ")) {
+                        System.out.println("    Please specify a deadline description and time using '/by'.");
+                        break;
```
Good usage of syntax usage check

asytrix99 reviewed (commented):
Just some nits - apply SLAP, good checks. Overall, LGTM!

Kurokishi592 committed:
Add simple text-ui-test

Kurokishi592 committed:
Refactor code to improve code quality and readability

Kurokishi592 committed:
Change logic to only operate on command words.

Add 'add' as one command, to retain level 2.
Any non-command are handled as the default for switch cases

Kurokishi592 committed:
Add proper error handling using exceptions.

No longer use print lines for error, use KurokishiException class instead.
All methods use exceptions too instead of printing messages directly.
Error handling is now consistent.

Kurokishi592 committed:
Merge branch 'branch_Level-5' into master

Daviancold commented on src/main/java/KurokishiException.java:
Do consider using a more informative name for your exception

Daviancold commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,239 @@
+            try {
+                if (command.equals("bye")) {
+                    System.out.println("[SYSTEM NOTICE] Session concluded. Probability of future contact: high. Glory to Humanity.");
+                    printDashLine();
```
Good use of helper method for repeated logic (printing new lines)!

Daviancold commented on src/main/java/Kurokishi.java:
```diff
@@ -0,0 +1,239 @@
+    }
+
+    private static void printLogo() {
+        String logo = " _  __  \n"
```
Perhaps could consider making a non-changed variable FINAL

Kurokishi592 committed:
Organise classes into packages

Kurokishi592 committed:
Refactor phase 1

Kurokishi.java main code is shortened significantly
Added new packages to better organise code: command, parser, ui
Completed Ui.java, TaskList.java and Command.java

Kurokishi592 committed:
Complete command package:

command package handles all the commands parsed by inputs.
Command.java is an interface, which all other specific command implements.

Kurokishi592 committed:
Finalise parser and main, refactored code runs as intended.

Kurokishi592 committed:
Merge branch 'refactor'
Merge refactored code into master, with packages, interfaces,
and exceptions for condensed and clean main logic.

Kurokishi592 committed:
Delete dead code of old main() logic after refactoring.

Current refactored code works after merging to master branch.
Old code is redundant, hence deleted.

Kurokishi592 committed:
Add level-6: delete

Kurokishi592 committed:
Add storage funtion under data package.

Kurokishi592 committed:
Integrate save feature

Kurokishi592 committed:
Add meta file

Kurokishi592 committed:
Merge branch 'branch-Level-6'
Add delete feature.

Kurokishi592 committed:
Merge branch 'branch-Level-7'
Add save and storage feature.

Kurokishi592 committed:
Integrate save feature and delete feature and fix bugs.

Kurokishi592 committed:
Add comments in javadoc format

Kurokishi592 committed:
Add Level 8: Date and Time.

Kurokishi can read date and time for inputs:
Deadline and Event can take in date only or date and time only or both
in the yyyy-MM-dd or yyyy-MM-dd HHmm format.

Event also compares /to time to be after /from time.

Kurokishi592 committed:
Merge pull request #1 from Kurokishi592/branch-Level-8

Level-8: Dates and Times

Kurokishi592 committed:
Fixed a small display issue for event on txt file

Kurokishi592 committed:
Add Level-9: Find a keyword from taskslist.

Kurokishi592 committed:
Merge pull request #2 from Kurokishi592/branch-Level-9

Level-9: Find

Kurokishi592 committed:
Add JavaDoc comments to the code.

Add header comments to all non-private classes/methods.

Kurokishi592 committed:
Merge pull request #3 from Kurokishi592/branch-JavaDoc

A-JavaDoc: Add JavaDoc comments

Kurokishi592 committed:
Add a User Guide

Kurokishi592 committed:
Fix User Guide

Kurokishi592 committed:
Fix User Guide Image

Kurokishi592 committed:
Fix Prompting Text Display

Kurokishi592 committed:
Minor Fix of Website by removing exclamation marks

Kurokishi592 committed:
Fix switch statement indentation

Kurokishi592 committed:
Fix run command and data storage path in README

Updated instructions for running the application and data storage location.
